### PR TITLE
feat: secure all function host endpoints

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description

This PR will add JWT token authentication to all function endpoints.

## References

* #1064

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            serviceCollection.AddScoped<IClaimsPrincipalAccessor, ClaimsPrincipalAccessor>();
+            serviceCollection.AddScoped<ClaimsPrincipalContext>();
+            serviceCollection.AddScoped(_ => new OpenIdSettings(metadataAddress, audience));
+        }
+
+        /// <summary>
+        /// Adds registration of ActorMiddleware, ActorContext and ActorProvider.
+        /// </summary>
+        /// <param name="container">ServiceCollection container</param>
+        public static void AddActorContext(this IServiceCollection container)
```
Comment (cmgrd) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/ServiceCollectionExtensions.cs:48]: Same type, different parameter names.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/MarketParticipants/IMarketParticipantRepository.cs [hunk 2]
```diff
@@ -27,6 +27,8 @@ public interface IMarketParticipantRepository
 
         Task<IReadOnlyCollection<MarketParticipant>> GetAsync(IEnumerable<Guid> ids);
 
+        Task<MarketParticipant> GetOrNullAsync(Guid id);
```
Comment (cmgrd) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/MarketParticipants/IMarketParticipantRepository.cs:30]: Shouldn't `MarketParticipant` be nullable with this naming?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs [hunk 3]
```diff
@@ -68,6 +71,11 @@ internal static void ConfigureServices(IServiceCollection serviceCollection)
             serviceCollection.AddScoped<FunctionTelemetryScopeMiddleware>();
             serviceCollection.AddScoped<MessageMetaDataMiddleware>();
             serviceCollection.AddScoped<FunctionInvocationLoggingMiddleware>();
+            serviceCollection.AddJwtTokenSecurity();
+            serviceCollection.AddActorContext();
+            serviceCollection.AddScoped(s => new ServiceBusActorContextMiddleware(
+                s.GetService<ILogger<ServiceBusActorContextMiddleware>>()!,
+                s.GetService<IActorContext>()!));
```
Comment (cmgrd) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs:78]: Why do we need to use GetService here? Can't we just depend on those services and let the container do its job?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs [hunk 4]
```diff
@@ -31,6 +31,13 @@ public class MarketParticipantRepository : IMarketParticipantRepository
             _chargesDatabaseContext = chargesDatabaseContext;
         }
 
+        public async Task<MarketParticipant> GetOrNullAsync(Guid id)
```
Comment (cmgrd) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Persistence/Repositories/MarketParticipantRepository.cs:34]: Nullable?

**Review by cmgrd — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs (hunk 3 above)
Comment (lasrinnil) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.FunctionHost/Configuration/SharedConfiguration.cs:78]: I am aware this implementation is not clean, but I tried other solutions with no luck. In the future, this registration will change.

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**